### ISSUE TITLE
@flow directive strips out docblock comments and messes up other comments

### ISSUE THREAD

**Issue by DmitrySoshnikov** [babel/babel#2022]
```js
/**
 * @providesModule Foo
 * @flow
 */
 
 /* other comment */
 
 module.exports = {};
```

Compiles to:

```js
"use strict";

module.exports = {};

/* other comment */
```

However, the:

```js
/**
 * @providesModule Foo
 */
 
 /* other comment */
 
 module.exports = {};
```

Correctly to:

```js
/**
 * @providesModule Foo
 */

/* other comment */

"use strict";

module.exports = {};
```

**Comment by sebmck** [comment]
This is due to changes done in #1934. So the `@flow` portion should just be removed from the contents or?

**Comment by DmitrySoshnikov** [comment]
There is actually even [test](https://github.com/babel/babel/tree/91fb64eaf53c67a7ed7605aab71f8dcaa4d39ae2/packages/babel/test/fixtures/transformation/flow/strip-directive) for this. What was the implications? We should not be loosing the dockblock, since `shouldPrintComment` is provided.

Yeah, we should probably be stripping only the `@flow` directive only in this case (by replacing it with a whitespace), but preserving the comments.

```js
/**
 * @providesModule Foo
 * @flow
 */
```

To:

```js
/**
 * @providesModule Foo
 *
 */
```

CC @leebyron

**Comment by leebyron** [comment]
Agreed. Only the `@flow` portion should be removed.

**Comment by DmitrySoshnikov** [comment]
Thank you!

**Comment by sebmck** [comment]
Sorry for the delay. Fixed as of 5.7.4, thanks!
